From 90be3d06287096932e950004a62b677f163684fc Mon Sep 17 00:00:00 2001
From: Abhishek Singh Chouhan <achouhan@apache.org>
Date: Wed, 6 Dec 2017 19:43:30 +0530
Subject: [PATCH 446/471] HBASE-19440 Not able to enable balancer with
 RSGroups once disabled

Signed-off-by: Andrew Purtell <apurtell@apache.org>

(cherry picked from commit 48b41c4b13fc4208d4f360e388a66dbe0ce533e3)

Change-Id: I69f35f410f1216f0cbb3e0239332246f502bc6db
Author: Abhishek Singh Chouha
Reason: Bug
Ref: CDH-65325
---
 .../apache/hadoop/hbase/rsgroup/TestRSGroups.java  |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroups.java b/hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroups.java
index 6e5148d..a6aca31 100644
--- a/hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroups.java
+++ b/hbase-rsgroup/src/test/java/org/apache/hadoop/hbase/rsgroup/TestRSGroups.java
@@ -18,6 +18,7 @@
 package org.apache.hadoop.hbase.rsgroup;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
@@ -270,4 +271,14 @@ public class TestRSGroups extends TestRSGroupsBase {
       }
     });
   }
+
+  @Test
+  public void testRSGroupBalancerSwitch() throws IOException {
+    //Balancer is initially off in the test, set to true and check
+    assertFalse(admin.setBalancerRunning(true, true));
+    assertTrue(admin.isBalancerEnabled());
+    //Set balancer off and check if it actually turned off
+    assertTrue(admin.setBalancerRunning(false,true));
+    assertFalse(admin.isBalancerEnabled());
+  }
 }
-- 
1.7.9.5

